### PR TITLE
docs: add terminology to manpage

### PR DESCRIPTION
**Pull request by Rylan12**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Closes https://github.com/Homebrew/brew/issues/11489

This PR adds the [terminology table](https://docs.brew.sh/Formula-Cookbook#homebrew-terminology) from the Formula Cookbook to the manpage. I've chosen to add it toward the bottom in a `TERMINOLOGY` section. I chose the bottom because I felt that the command descriptions should be first, but if there's a better place for it I can move.

I've also converted from table format to list format because tables don't seem to be doable in the manpage (or, at least, `ronn` doesn't convert them).

At the moment, this just copies directly from the formula cookbook but this can be converted to its own list if we think that makes more sense.

### CODE REVIEW

File: docs/Manpage.md [hunk 1]
```diff
@@ -2052,6 +2052,24 @@ And for an authenticated HTTP proxy:
 
     export http_proxy=http://$USER:$PASSWORD@$HOST:$PORT
 
+## TERMINOLOGY
+
+**Formula**: The package definition (e.g. `/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/foo.rb`)
+
+**Keg**: The installation prefix of a **Formula** (e.g. `/usr/local/Cellar/foo/0.1`)
+
+**opt prefix**: A symlink to the active version of a **Keg** (e.g. `/usr/local/opt/foo`)
```
Comment (MikeMcQuaid) [inline docs/Manpage.md:2061]: Think we can omit this given the manpage doesn't reference it.

File: docs/Manpage.md [hunk 2]
```diff
@@ -2052,6 +2052,24 @@ And for an authenticated HTTP proxy:
 
     export http_proxy=http://$USER:$PASSWORD@$HOST:$PORT
 
+## TERMINOLOGY
+
+**Formula**: The package definition (e.g. `/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/foo.rb`)
+
+**Keg**: The installation prefix of a **Formula** (e.g. `/usr/local/Cellar/foo/0.1`)
```
Comment (MikeMcQuaid) [inline docs/Manpage.md:2059]: Keg-only might be worth defining (in both places) as the manpage references it

File: docs/Manpage.md [hunk 3]
```diff
@@ -2052,6 +2052,24 @@ And for an authenticated HTTP proxy:
...
+
+**Cellar**: All **Kegs** are installed here (e.g. `/usr/local/Cellar`)
+
+**Tap**: A Git repository of **Formulae** and/or commands (e.g. `/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core`)
+
+**Bottle**: Pre-built **Keg** used instead of building from source (e.g. `qt-4.8.4.catalina.bottle.tar.gz`)
+
+**Cask**: An extension of Homebrew to install macOS native apps (e.g. `/Applications/MacDown.app/Contents/SharedSupport/bin/macdown`)
+
+**Brew Bundle**: An extension of Homebrew to describe dependencies (e.g. `brew 'myservice', restart_service: true`)
```
Comment (MikeMcQuaid) [inline docs/Manpage.md:2071]: Don't think this is necessary given the longer explanation in the manpage.

File: docs/Manpage.md [hunk 4]
```diff
@@ -2052,6 +2052,24 @@ And for an authenticated HTTP proxy:
 
     export http_proxy=http://$USER:$PASSWORD@$HOST:$PORT
 
+## TERMINOLOGY
```
Comment (MikeMcQuaid) [inline docs/Manpage.md:2055]: I think ideally we'd have this be in the manpage before we use these terms.

File: docs/Manpage.md [hunk 5]
```diff
@@ -2052,6 +2052,24 @@ And for an authenticated HTTP proxy:
...
+
+**Formula**: The package definition (e.g. `/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/foo.rb`)
```
Comment (MikeMcQuaid) [inline docs/Manpage.md:2057]: Think the `e.g.`s can be skipped in the manpage

**Review by MikeMcQuaid — COMMENTED** [review]
Nice work here so far! I think it's worth just duplicating the terminology as I think the format/terms we want are a little different and the audiences are too (people using Homebrew vs. people creating new formulae for Homebrew).

File: docs/Manpage.md (hunk 4 above)
Comment (Rylan12) [reply docs/Manpage.md:2055]: Right at the beginning after the `DESCRIPTION` section?

File: Library/Homebrew/manpages/brew.1.md.erb [hunk 6]
```diff
@@ -28,6 +28,24 @@ Homebrew is the easiest and most flexible way to install the UNIX tools Apple
...
+
+**Keg-only**: A **Formula** is **Keg-only** if it is not linked into the Homebrew prefix
+
+**Cellar**: All **Kegs** are installed here
+
+**Caskroom**: All **Casks** are installed here
+
+**Tap**: A Git repository of **Formulae** and/or commands
+
+**Bottle**: Pre-built **Keg** used instead of building from source
```
[the remark this answers is not in the document]
Comment (Rylan12) [reply Library/Homebrew/manpages/brew.1.md.erb:47]: These look good to me

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks @Rylan12!

**Pull request merged**